### DISCUSSION TITLE
Allow inline variable declaration for in and ref parameter types

### DISCUSSION THREAD

**Discussion by gafter** [dotnet/csharplang#1289] [General]
@Kentalot commented on [Mon Jan 29 2018](https://github.com/dotnet/roslyn/issues/24525)

**Version Used**: 
I'm using C# 7.2 and version 15.5.3 of VS 2017 Enterprise.

**Steps to Reproduce**:

``` C#
private static void GetValue(out string input)
{
    input = null;
}
private static void DoNothing(ref string input)
{
}

private static void DoNothingIn(in string input)
{
}

private static void Call()
{
    GetValue(out var x);
    DoNothing(ref var y = x);
    DoNothingIn(in var z = null);
}
```

**Expected Behavior**: Compiles and y is usable after calling DoNothing and z is usable after calling DoNothingIn and all of x, y, and z are basically null;

**Actual Behavior**:
Does not compile since you can't inline declare variables for ref or in keywords.  I hope that's clear.  Thanks!

---

@HaloFour commented on [Mon Jan 29 2018](https://github.com/dotnet/roslyn/issues/24525#issuecomment-361463569)

Proposals to change the behavior of the language are taken at the [C# Language](https://github.com/dotnet/csharplang/) repository.

**Comment by MkazemAkhgary** [comment]
Also #1077.

**Comment by Kentalot** [comment]
Thanks guys.  Sorry for posting in the wrong repo.  #1077 is similar for the ref part, but I guess mine is new-ish that it's requesting a similar thing for the in keyword (I know they're related).  But anyway, thanks for considering my request!

**Comment by MkazemAkhgary** [comment]
@ghost symmetry for `out` would be post assignment.

    public bool M1(out Foo x)
    {
        return M2(x = out Bar y);
    }

pre assignment is not possible, because methods with `out` parameter are always forced to initialize `out` parameters which leads to overwriting your default values. its a completely unexpected behavior.
